lisa/operating_system.py

```diff
@@ -179,7 +245,50 @@ def install_packages(
         if self._first_time_installation:
             self._first_time_installation = False
             self._initialize_package_installation()
-        self._install_packages(package_names)
+
+        self._install_packages(package_names, signed)
+
+    # query_package
+    # Query if a package/tool is installed on the node.
+    # Return Value - bool
+    def query_package(
+        self, package: Union[str, Tool, Type[Tool]], signed: bool = True
+    ) -> bool:
+        if isinstance(package, str):
+            package_name = package
+        elif isinstance(package, Tool):
+            package_name = package.package_name
+        else:
+            assert isinstance(package, type), f"actual:{type(package)}"
+            # Create a temp object, it doesn't query.
+            # So they can be queried together.
+            tool = package.create(self._node)
+            package_name = tool.package_name
+
+        return self._query_package(package_name)
+
+    # TODO: Implement update_packages
+    def update_packages(
+        self, packages: Union[str, Tool, Type[Tool], List[Union[str, Tool, Type[Tool]]]]
+    ) -> None:
+        pass
+
+    def _get_os_version(self) -> OsVersion:
+        cmd_result = self._node.execute(cmd="cat /etc/os-release", no_error_log=True)
+        if cmd_result.exit_code == 0 and cmd_result.stdout != "":
+            for row in cmd_result.stdout.split("\n"):
+                if "=" in row:
```

Comment:
use regular expression is clearer and clean

Reply:
Done